lib/plugins.js

```diff
@@ -30,18 +33,35 @@ function mergeByName(acceptor, donor, name, onConflict) {
   }, this);
 }
 
-var plugins = [],
-    dirlist;
+var cachedPlugins = {};
 
-try {
-  dirlist = fs.readdirSync(CWD);
-} catch (err) {
-  dirlist = [];
+function _setMockPlugins(pluginPath, plugins) {
+  cachedPlugins[pluginPath] = plugins;
 }
-  
-dirlist.reduce(function(plugins, subdir) {
-    var _path = path.resolve(CWD, subdir, 'manifest.json'),
-        manifest;
+
+function getPlugins(config, ignoreCache) {
```

Comment:
Yes, you could call `getPlugins({ plugins: true, pluginPath: '/foo/bar' }, true)` and it will ignore the cached version and read from the directory again.

I added a separate argument so people don't have to extend or mutate the config object.